dataland-frontend/src/App.vue

```diff
@@ -1,68 +1,131 @@
 <template>
+  <DynamicDialog />
   <router-view />
 </template>
 
 <script lang="ts">
 import Keycloak from "keycloak-js";
+import DynamicDialog from "primevue/dynamicdialog";
 import { computed, defineComponent } from "vue";
+import { logoutAndRedirectToUri } from "@/utils/KeycloakUtils";
+import {
+  SessionDialogMode,
+  startSessionSetIntervalFunctionAndReturnItsId,
+  updateTokenAndItsExpiryTimestampAndStoreBoth,
+} from "@/utils/SessionTimeoutUtils";
+import SessionDialog from "@/components/general/SessionDialog.vue";
+import { KEYCLOAK_INIT_OPTIONS } from "@/utils/Constants";
+import { useSharedSessionStateStore } from "@/stores/stores";
 
 export default defineComponent({
   name: "app",
+  components: { DynamicDialog },
+
   data() {
     return {
-      keycloak: undefined,
-      keycloakPromise: undefined,
+      keycloakPromise: undefined as undefined | Promise<Keycloak>,
+      resolvedKeycloakPromise: undefined as undefined | Keycloak,
       keycloakAuthenticated: false,
+      functionIdOfSessionSetInterval: undefined as number | undefined,
+    };
+  },
+
+  watch: {
+    currentRefreshTokenInSharedStore(newRefreshToken: string) {
+      if (this.resolvedKeycloakPromise && newRefreshToken) {
+        this.resolvedKeycloakPromise.refreshToken = newRefreshToken;
+        clearInterval(this.functionIdOfSessionSetInterval);
+        const openSessionWarningModalBound = this.openSessionWarningModal.bind(this);
+        this.functionIdOfSessionSetInterval = startSessionSetIntervalFunctionAndReturnItsId(
+          this.resolvedKeycloakPromise,
+          openSessionWarningModalBound
+        );
+      }
+    },
+  },
+
+  computed: {
+    currentRefreshTokenInSharedStore() {
+      return useSharedSessionStateStore().refreshToken;
+    },
+  },
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
```

Comment:
Check if this is truely required

Reply:
I also tried getting rid of this warning / suppression for quite a bit and couldn't do it.
Doing some configuration stuff in eslint could solve it: https://github.com/typescript-eslint/typescript-eslint/issues/4619#issuecomment-1055614155 

Leaving the code as is and disabling this warning here is in my opinion correct. Using async before created() is legitimate code and it would be very obfuscated to write something here that works around the ESLINT complaint so instead we should just disable it.

Reply:
I disagree. The function definition does not specify that it is possible to return a promise. And Experiments with a (very long) timeout reveal that the returned promise is never awaited. Therefore, this function should not return a promise as ESLint suggests. Using `this.keycloakPromise.then((keycloak) => {})` instead of the `async` modifier should be sufficient to achieve the desired behavior.

Reply:
Our adjustments to do this broke the feature.
I have currently no time for a fix due to higher priority of Marcins PR.